src/main/java/org/sonarlint/intellij/actions/SonarAnalyzeFilesAction.java

```diff
@@ -33,44 +34,50 @@
 import org.sonarlint.intellij.trigger.SonarLintSubmitter;
 import org.sonarlint.intellij.trigger.TriggerType;
 import org.sonarlint.intellij.ui.SonarLintConsole;
-import org.sonarlint.intellij.util.SonarLintAppUtils;
 import org.sonarlint.intellij.util.SonarLintUtils;
 
-public class SonarAnalyzeEditorFileAction extends AbstractSonarAction {
-  public SonarAnalyzeEditorFileAction() {
+public class SonarAnalyzeFilesAction extends AbstractSonarAction {
+  public SonarAnalyzeFilesAction() {
     super();
   }
 
-  public SonarAnalyzeEditorFileAction(@Nullable String text, @Nullable String description, @Nullable Icon icon) {
+  public SonarAnalyzeFilesAction(@Nullable String text, @Nullable String description, @Nullable Icon icon) {
     super(text, description, icon);
   }
 
   @Override
-  protected boolean isEnabled(Project project, SonarLintStatus status) {
+  protected boolean isEnabled(AnActionEvent e, Project project, SonarLintStatus status) {
     if (status.isRunning()) {
       return false;
     }
 
-    return SonarLintUtils.get(SonarLintAppUtils.class).getSelectedFile(project) != null;
+    VirtualFile file = e.getData(CommonDataKeys.VIRTUAL_FILE);
+
+    // TODO ideally we would already check if the files are analyzable
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135)